function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautLambdaContainerHandler.java

```diff
@@ -399,48 +415,151 @@ private MicronautAwsProxyResponse<?> toAwsProxyResponse(
+            Optional.empty();
+    }
+
+    @NonNull
+    private Optional<Object> decodeRequestBody(@NonNull MicronautAwsProxyRequest<?> containerRequest,
+                                               @NonNull RouteMatch<?> finalRoute) {
+        return mediaTypeBodyDecoder.entrySet()
+            .stream()
+            .map(e -> decodeRequestBody(containerRequest, finalRoute, e))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .findFirst();
+    }
+
+    private static Optional<Object> decodeRequestBody(@NonNull MicronautAwsProxyRequest<?> containerRequest,
+                                                      @NonNull RouteMatch<?> finalRoute,
+                                                      @NonNull Map.Entry<MediaType, BiFunction<Argument<?>, String, Optional<Object>>> entry) {
+        return parseUndecodedBody(containerRequest, finalRoute, entry.getKey())
+            .flatMap(body -> decodeRequestBody(body, entry.getValue(), finalRoute));
+    }
+
+    @NonNull
+    private static Optional<Object> decodeRequestBody(@NonNull String body,
+                                               @NonNull BiFunction<Argument<?>, String, Optional<Object>> function,
+                                               @NonNull RouteMatch<?> finalRoute) {
+        if (StringUtils.isNotEmpty(body)) {
+            Argument<?> bodyArgument = parseBodyArgument(finalRoute);
+            return function.apply(bodyArgument, body);
+        }
+        return Optional.empty();
+    }
+
+    @Nullable
+    private static Argument<?> parseBodyArgument(@NonNull RouteMatch<?> finalRoute) {
+        Argument<?> bodyArgument = finalRoute.getBodyArgument().orElse(null);
+        if (bodyArgument == null) {
+            if (finalRoute instanceof MethodBasedRouteMatch) {
+                bodyArgument = Arrays.stream(((MethodBasedRouteMatch) finalRoute).getArguments())
+                    .filter(arg -> HttpRequest.class.isAssignableFrom(arg.getType()))
+                    .findFirst()
+                    .flatMap(TypeVariableResolver::getFirstTypeVariable).orElse(null);
             }
         }
+        if (bodyArgument != null) {
+            final Class<?> rawType = bodyArgument.getType();
+            if (Publishers.isConvertibleToPublisher(rawType) || HttpRequest.class.isAssignableFrom(rawType)) {
+                bodyArgument = bodyArgument.getFirstTypeVariable().orElse(Argument.OBJECT_ARGUMENT);
+            }
+        }
+        return bodyArgument;
+    }
+
+    @NonNull
+    private Optional<Object> getFormUrlEncodedDecodedBody(@Nullable Argument<?> bodyArgument,
+                                                          @NonNull String body) {
+        if (bodyArgument == null) {
+            return Optional.empty();
+        }
+        if (nestedBody(bodyArgument)) {
+            return Optional.of(formUrlEncodedBodyToConvertibleValues(body));
+        }
+        return bindFormUrlEncoded(bodyArgument, body);
```

Comment:
if Body is not nested), we are transforming the body into a map and then using `BeanPropertyBinder` to bind map to POJO.